app/src/main/java/com/github/bgabriel998/softwaredevproject/CameraPreview.java

```diff
@@ -2,63 +2,173 @@
 
 import android.content.Context;
 import android.content.pm.PackageManager;
+import android.content.res.Configuration;
+import android.graphics.Bitmap;
 import android.hardware.camera2.CameraAccessException;
 import android.hardware.camera2.CameraCharacteristics;
 import android.hardware.camera2.CameraManager;
+import android.hardware.display.DisplayManager;
+import android.net.Uri;
+import android.os.Bundle;
+import android.os.Handler;
+import android.os.Looper;
+import android.os.Message;
 import android.util.DisplayMetrics;
 import android.util.SizeF;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.Toast;
 
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
 import androidx.camera.core.AspectRatio;
 import androidx.camera.core.CameraSelector;
 import androidx.camera.core.ImageAnalysis;
+import androidx.camera.core.ImageCapture;
+import androidx.camera.core.ImageCaptureException;
 import androidx.camera.core.ImageProxy;
 import androidx.camera.core.Preview;
 import androidx.camera.lifecycle.ProcessCameraProvider;
 import androidx.camera.view.PreviewView;
+import androidx.constraintlayout.widget.ConstraintLayout;
 import androidx.core.content.ContextCompat;
 import androidx.core.util.Pair;
+import androidx.fragment.app.Fragment;
 import androidx.lifecycle.LifecycleOwner;
 
 import com.google.common.util.concurrent.ListenableFuture;
 
+import java.io.File;
 import java.util.concurrent.ExecutorService;
 import java.util.concurrent.Executors;
 
 /**
- * CameraPreview is used to display the camera-preview of the phone on a view.
+ * A {@link Fragment} subclass that represents the camera-preview.
+ * Use the {@link CameraPreview#newInstance} factory method to
+ * create an instance of this fragment.
+ * See: https://github.com/android/camera-samples/blob/main/CameraXBasic
  */
-public class CameraPreview{
+public class CameraPreview extends Fragment{
 
-    private final PreviewView previewView;
+    private PreviewView previewView;
     private ProcessCameraProvider cameraProvider;
-    private final ExecutorService cameraExecutor;
-    private final Context context;
+    private ImageCapture imageCapture;
+    private ExecutorService cameraExecutor;
+    private Context context;
+
+    private final String FILENAME = "yyyy-MM-dd-HH-mm-ss-SSS";
+    private final String PHOTO_EXTENSION = ".jpg";
+    private final int PICTURE_TAKEN = 1;
+    private final int FAILED_TO_TAKE_PICTURE = 0;
+    private final int FILE_LENGTH = 27;
+
+    private ImageAnalysis imageAnalysis;
+
+    private int previewDisplayId = -1;
+
+    private DisplayManager displayManager;
+
+    private DisplayManager.DisplayListener displayListener;
+
+    String lastToast = null;
```

Comment:
Make attribute private since getter and setter methods exist
```suggestion
    private String lastToast = null;
```

Reply:
There are no getter and setter methods, I am directly calling CameraPreview.lastToat from Button1Activity.